### PR TITLE
Add case-insensitive/CI instance

### PR DESCRIPTION
**Pull request by phadej**
case-insensitive is in Haskell Platform!

### CODE REVIEW

File: quickcheck-instances.cabal [hunk 1]
```diff
@@ -39,6 +39,7 @@ Library
 
                        array >= 0.3 && < 0.6,
                        bytestring >= 0.9 && < 0.11,
+                       case-insensitive >=1.2.0.4 && <1.3,
```
Comment (phadej) [inline quickcheck-instances.cabal]: lower bounds ~arbitrary, chose one from Stackage LTS-2.22

**Comment by aslatter** [comment]
The `CI` type seems like an odd one, where the `Eq` and `Ord` instances can't distinguish case, but you can still write code that can tell the difference between `CI.mk "foo"` and `CI.mk "FOO"`.

So I guess the `CoArbitrary` instance is okay? But it does seem weird.

**Comment by phadej** [comment]
@aslatter I have no idea why `original` is exposed, there is an issue about moving it to `Unsafe` part.

**Comment by aslatter** [comment]
Well, if it ends up moved to `Unsafe` then it might make sense to use the case-folded version of `CI` to perturb the generator.

But until then I think this is good.

**Pull request merged**